lib/improver/tests/ensemble_calibration/ensemble_calibration/test_EnsembleCalibration.py

```diff
@@ -186,210 +178,335 @@ def test_basic_wind_speed_realizations(self):
+        super().setUp()
+        self.expected_specific_gaussian_predictor_data = np.array(
+            [[274.1325, 275.0439, 275.2852],
+             [277.02405, 277.4005, 278.3493],
+             [280.0655, 280.34113, 281.23315]], dtype=np.float32)
+
+        self.expected_specific_gaussian_variance_data = np.array(
+            [[0.9980779, 1.0092506, 0.06006714],
+             [0.11590514, 0.10101636, 0.06006713],
+             [0.27223495, 0.01540309, 0.00423481]], dtype=np.float32)
+
+        self.expected_specific_truncated_gaussian_predictor_data = np.array(
+            [[0.8932284, 1.6185861, 2.3541362],
+             [3.8048353, 4.418258, 5.1315794],
+             [6.3740964, 6.950509, 7.6758633]], dtype=np.float32)
+
+        self.expected_specific_truncated_gaussian_variance_data = np.array(
+            [[1.6119667, 1.6299378, 0.10240279],
+             [0.1922579, 0.16829637, 0.10240279],
+             [0.4438519, 0.0305187, 0.01254779]], dtype=np.float32)
+
+    @ManageWarnings(
+        ignored_messages=IGNORED_MESSAGES, warning_types=WARNING_TYPES)
+    def test_gaussian_realizations_data_check(self):
         """
-        Test that the plugin returns the calibrated predictor and the
-        calibrated variance if an alternative name for the calibration
-        is provided. The ensemble mean is the predictor.
+        Test that the plugin returns an iris.cube.CubeList
+        of temperature cubes with the expected data, where the plugin
+        returns a cubelist of, firstly, the predictor and, secondly the
+        variance. The ensemble realizations is the predictor.
         """
-        calibration_method = "nonhomogeneous gaussian regression"
         distribution = "gaussian"
-        desired_units = "degreesC"
-        plugin = Plugin(calibration_method, distribution, desired_units)
+        predictor_of_mean_flag = "realizations"
+        plugin = Plugin(
+            self.calibration_method, distribution,
+            predictor_of_mean_flag=predictor_of_mean_flag)
         calibrated_predictor, calibrated_variance = plugin.process(
             self.current_temperature_forecast_cube,
             self.historic_temperature_forecast_cube,
             self.temperature_truth_cube)
-        self.assertIsInstance(calibrated_predictor, iris.cube.Cube)
-        self.assertIsInstance(calibrated_variance, iris.cube.Cube)
+        self.assertArrayAlmostEqualLowerPrecision(
+            calibrated_predictor.data,
+            self.expected_specific_gaussian_predictor_data, decimal=4)
+        self.assertArrayAlmostEqualLowerPrecision(
+            calibrated_variance.data,
+            self.expected_specific_gaussian_variance_data)
+        self.assertArrayAlmostEqual(
+            calibrated_predictor.data,
+            self.expected_gaussian_mean_data, decimal=0)
+        self.assertArrayAlmostEqual(
+            calibrated_variance.data,
+            self.expected_gaussian_variance_data, decimal=0)
```

Comment:
I've corrected the level of precision specified for the input data. Note that for these tests, I've made two comparisons, one versus the output expected when using the ensemble realizations as a predictor, and another comparison versus when the ensemble mean is the predictor. This comparison is just to show that the outputs from using the ensemble mean and using the ensemble realizations as the predictor are broadly similar e.g. to 0 decimal places, but it would not be expected that these outputs would match specifically as the inputs to the ensemble calibration are different i.e. ensemble mean as input and ensemble realizations as input. This assertion is just trying to show that the results are broadly similar.